api/hasura/actions/create_circle.ts

```diff
@@ -0,0 +1,173 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { z } from 'zod';
+
+import { gql } from '../../../api-lib/Gql';
+import {
+  profiles_constraint,
+  order_by,
+  profiles_update_column,
+} from '../../../src/lib/gql/zeusHasuraAdmin';
+import { createCircleSchemaInput } from '../../../src/lib/zod';
+import { MutationCreate_CircleArgs } from '../customTypes/createCircleCustomTypes';
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
+  const { object }: MutationCreate_CircleArgs = req.body.input;
+  try {
+    createCircleSchemaInput.parse(object);
+  } catch (err) {
+    if (err instanceof z.ZodError) {
+      return res.status(422).json({
+        extensions: err.issues,
+        message: 'Invalid input',
+        code: '422',
+      });
+    }
+  }
+  const address = object.address.toLowerCase();
```

Comment:
The zod parser should probably be rewritten so that the address is included and that lower case is also done with it. I can help with that.

Reply:
i've left the tolowercase for now